### ISSUE TITLE
No 404 with server side rendering 

### ISSUE THREAD

**Issue by patrickweegen** [pkellner/pluralsight-course-react-aspnet-core#3]
Hi!

I just done the module "Building a Custom Webpack Configuration for React and Core".

When I tried if a 404 gets returned when entering an invalid URL into the browser, I always got 200 and no 404.
After some digging into the code I found out that with your implemntation `rendererInstance.routestatus` is  always `undefined` when it gets checked in `Server.js`.

```javascript
app.get('*', (req, res) => {
    const rendererInstance = Renderer(req);

    // rendererInstance is always undefined here
    if (rendererInstance.routestatus == 404) {
        res.status(404).end("Not found, 404 status returned");
    } else {
        res.send(Renderer(req).htmlcode);
    }
});
```

The problem is that because of the default route `<Route render={() => <RouteNotFound />} />` is existing the router does always get a result.
I have tired to implement it like shown in the docs: [click me](https://github.com/ReactTraining/react-router/blob/master/packages/react-router-dom/docs/guides/server-rendering.md#404-401-or-any-other-status)

You can find the changed files [here](https://gist.github.com/patrickweegen/d096861b81957e88a69f27bcda8a38ab)
With this solution, the server returns a 404 when an invalid request was given and also returns the styled error page.

I think this would be worth to get updated in the course.

**Comment by pkellner** [comment]
Hi @patrickweegen again. I'm very sorry you had to go through what you did to figure out what was wrong.  Your solution is perfect and very well thought out and researched. As you say, 404's are not returned as they should be.

It turns out (english expression for "my bad"), I did solve the problem starting in module 5 by changing the RouteNotFound.js file to do exactly what you did in the Routes.js file.  [See here](https://github.com/pkellner/pluralsight-course-react-aspnet-core/blob/master/m5-use-rest-services/reactapp/ClientApp/RouteNotFound.js)

Since I never showed the RouteNotFound.js code in the course (that I can find right now), there would be know way to know that I had done that hence you had to do all that work to figure it out. I'm going to update the module 2 code now so no one else runs into this.

Thanks for letting me know. That's very helpful and I'm sure everyone else going through the course will appreciate your efforts.

Best Regards, -Peter

**Comment by pkellner** [comment]
I've updated the source now for all modules that had the problem.

**Comment by patrickweegen** [comment]
Hi @pkellner 
Thanks for updating. SInce I'm rather new to all this React and JavaScript thing, it was also a little fun to hunt the Bug ;-)

I'll close the Issue.

**Comment by pkellner** [comment]
@patrickweegen Good to hear. I'm the same way.  I'm like a dog with a bone when something does not work as I expect.  On a side note, not to shamelessly plug my new course on server-side React, but I did learn a lot researching and building the app in this course. I had looked at NextJS before going down the path here, but at the time I was writing this course, it did not seem mature enough. Now it is IMHO as of V7. If you do plan on building production server-side React, I'd take a look at it before going down the path I outline in this course, but instead look at the [code and associated video here]([url](https://github.com/pkellner/pluralsight-course-server-side-rendered-react-nextjs))   The kind of problem I created for you and you solved, simply don't exist as it's baked into the framework. There are limitations though so nothing is perfect.
